crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -0,0 +1,174 @@
+use crate::archetype::ArchetypeComponentId;
+use crate::archetype::ArchetypeComponentInfo;
+use crate::component::{ComponentId, ComponentTicks, Components, StorageType};
+use crate::storage::{Column, SparseSet};
+use bevy_ptr::{OwningPtr, Ptr, PtrMut, UnsafeCellDeref};
+use std::cell::UnsafeCell;
+
+/// The type-erased backing storage and metadata for a single resource within a [`World`].
+///
+/// [`World`]: crate::world::World
+pub struct ResourceData {
+    column: Column,
```

Comment:
What's the reasoning for using a unit-length `Column`, instead of something with `Option`?

Reply:
the closest type for what we want is `Option<Box<dyn Component>>` + changeticks I think. unfortunately we cant actually use that because bevy supports/will support components/resources that dont have rust types which is incompatible with `dyn Component`. I assume `Column` was chosen because its the only thing we've already got implemented that manages a heap allocation without specifying a type

Reply:
Makes sense. Perhaps a more specialized type could be used in a future PR, if it ends up being faster.